frontend/src/CreateDiagram/CreateDiagram.jsx

```diff
@@ -2,27 +2,51 @@ import React, { useState } from "react";
 
 import { Link } from "react-router-dom";
 
+import "./CreateDiagram.scss";
+
 export default function CreateDiagram({ onSubmit }) {
   const [diagramName, setDiagramName] = useState("");
 
   function submit() {
     onSubmit({ diagramName });
   }
 
+  function isValid() {
+    // TODO: use a regex for this
+    if (diagramName.includes("-")) {
+      return false;
+    }
+    if (diagramName.length < 1) {
+      return false;
+    }
+    return true;
+  }
+
+  function onDiagramNameChange(e) {
+    const newText = e.target.value.replace(/[^a-z0-9_]/gi, "");
```

Comment:
I think this will replace all upper case letters too, which probably isn't the desired behaviour. I think you said that this changes in the next PR though...